error on build samples in vs2010

- OpenCV version: 3.x
- Windows XP x86
CMake: GCC 3.4.3

Error on build samples:
"Q:\ocv\opencv\build_x86_pre\OpenCV.sln" (целевой объект по умолчанию) (1) ->
"Q:\ocv\opencv\build_x86_pre\ALL_BUILD.vcxproj.metaproj" (целевой объект по умо
лчанию) (2) ->
"Q:\ocv\opencv\build_x86_pre\samples\cpp\example_detect_mser.vcxproj.metaproj"
(целевой объект по умолчанию) (86) ->
"Q:\ocv\opencv\build_x86_pre\samples\cpp\example_detect_mser.vcxproj" (целевой
объект по умолчанию) (87) ->
(Целевой объект ClCompile) ->
  ..\..\..\samples\cpp\detect_mser.cpp(285): error C2668: 'acos' : ambiguous ca
ll to overloaded function [Q:\ocv\opencv\build_x86_pre\samples\cpp\example_dete
ct_mser.vcxproj]

After replaceent
`float	pi = (float)acos(-1);` 
in detect_mser.cpp(285) to
`float	pi = (float)acos(static_cast<float>(-1));`
the build was successful.

--- a/samples/cpp/detect_mser.cpp
+++ b/samples/cpp/detect_mser.cpp
@@ -187,7 +187,7 @@ static void onMouse(int event, int x, int y, int flags, void*)
         else
             rObs -= (float)0.1;
     }
-    float pi = (float)acos(-1.0);
+    float pi = static_cast<float>(CV_PI);
     if (thetaObs>pi)
     {
         thetaObs = -2 * pi + thetaObs;
@@ -282,7 +282,7 @@ static void DrawOpenGLMSER(Mat img, Mat result)
             break;
         if (key == 0x20)
             rotateEnable = !rotateEnable;
-        float	pi = (float)acos(-1);
+        float pi = static_cast<float>(CV_PI);
 
         switch (key) {
             case '5':

